### PR TITLE
feat: add custom npm registry when installing dependencies

### LINKED ISSUE #972: feat: provide npm registry customization
zx has internal feature to install package (see `deps.ts` and `cli.ts`), but the registry path is hardcoded. Let's make it configurable via the CLI opts too.

```
zx --install-registry='https://npm-proxy.example.com' script.mjs
```

### PR DESCRIPTION
**Pull request by easymikey**
### Description
Add custom npm registry with composition install and registry flags.

Fixes #972

```js
zx --install --registry='https://npm-proxy.example.com' script.mjs
```
Screenshot 2024-12-16 at 15 41 17

- [x] Tests pass
- [ ] Appropriate changes to README are included in PR

### CODE REVIEW

**Comment by easymikey** [comment]
@antongolub you can start the review:)

File: src/cli.ts [hunk 1]
```diff
@@ -51,26 +51,35 @@ export function printUsage() {
...
+   --quiet                     suppress any outputs
+   --verbose                   enable verbose mode
+   --shell=<path>              custom shell binary
+   --prefix=<command>          prefix all commands
+   --postfix=<command>         postfix all commands
+   --cwd=<path>                set current directory
+   --eval=<js>, -e             evaluate script
+   --ext=<.mjs>                default extension
+   --install, -i               install dependencies
+   --install-registry<path>    install install dependencies via custom npm registry URL
```
Comment (antongolub) [inline src/cli.ts:63]: --install-registry=<path> install registry URL

File: src/cli.ts [hunk 2]
```diff
@@ -208,6 +217,12 @@ export async function importPath(
     const deps = parseDeps(await fs.readFile(filepath))
     await installDeps(deps, dir)
   }
+
```
Comment (antongolub) [inline src/cli.ts:220]: Let's just modify the current flow:
```js
if (argv.install) {
    const deps = parseDeps(await fs.readFile(filepath))
    await installDeps(deps, dir, argv['install-registry'])
}
```

**Review by antongolub — CHANGES_REQUESTED** [review]

File: src/cli.ts (hunk 2 above)
Comment (easymikey) [reply src/cli.ts:220]: Are you suggesting to make such a flow? Maybe then replace --install-registry with --registry as in npm?

```bash
zx --install --registry='https://npm-proxy.example.com' script.mjs
```

Comment (antongolub) [reply src/cli.ts:220]: ok

Comment (easymikey) [reply src/cli.ts]: d1

File: src/cli.ts (hunk 1 above)
Comment (easymikey) [reply src/cli.ts]: d1

File: src/cli.ts [hunk 3]
```diff
@@ -51,26 +51,27 @@ export function printUsage() {
...
+   --quiet                          suppress any outputs
+   --verbose                        enable verbose mode
+   --shell=<path>                   custom shell binary
+   --prefix=<command>               prefix all commands
+   --postfix=<command>              postfix all commands
+   --cwd=<path>                     set current directory
+   --eval=<js>, -e                  evaluate script
+   --ext=<.mjs>                     default extension
+   --install, -i                    install dependencies
+   --registry=<npm registry URL>    custom npm registry URL dependencies works with install flag
```
Comment (antongolub) [inline src/cli.ts:63]: I'd suggest:
`--registry=<URL>    npm registry, defaults to https://registry.npmjs.org/`

Update also `man/zx.1`

File: src/cli.ts [hunk 4]
```diff
@@ -51,26 +51,27 @@ export function printUsage() {
...
-   --postfix=<command>  postfix all commands
-   --cwd=<path>         set current directory
-   --eval=<js>, -e      evaluate script
-   --ext=<.mjs>         default extension
-   --install, -i        install dependencies
-   --version, -v        print current zx version
-   --help, -h           print help
-   --repl               start repl
-   --experimental       enables experimental features (deprecated)
+   --quiet                suppress any outputs
```
Comment (antongolub) [inline src/cli.ts:54]: Now we can keep the formatting here for git blame.

**Review by antongolub — CHANGES_REQUESTED** [review]
One more suggestion.

File: src/cli.ts (hunk 3 above)
Comment (easymikey) [reply src/cli.ts]: Cool suggest!

I update  `man/zx.1` and description for cli

File: src/cli.ts (hunk 4 above)
Comment (easymikey) [reply src/cli.ts]: Thanks to your suggestion. Revert formatting

**Review by antongolub — APPROVED** [review]
Seems fine. Thanks for the improvement!

**Pull request merged**